core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java

```diff
@@ -160,7 +160,7 @@ public void serialize(JsonTelemetryDataSerializer writer) throws IOException {
         tmp.setBaseData(getData());
         tmp.setBaseType(this.getBaseTypeName());
         envelope.setData(tmp);
-        envelope.setTime(LocalStringsUtils.getDateFormatter().format(getTimestamp()));
+        if (getTimestamp() != null) envelope.setTime(LocalStringsUtils.getDateFormatter().format(getTimestamp());
```

Comment:
missing closing parenthesis
```suggestion
        if (getTimestamp() != null) envelope.setTime(LocalStringsUtils.getDateFormatter().format(getTimestamp()));
```